src/ploomber_core/telemetry/telemetry.py

```diff
@@ -278,8 +280,9 @@ def check_dir_exist(input_location=None):
 
     p = p.expanduser()
 
-    if not p.exists():
-        p.mkdir(parents=True)
+    if not (cls is not None and cls.writable is False):
```

Comment:
Suggestion:
Instead of passing `cls` and checking if it's writable, we can add the parameter `create` with a default value of `True`.

Something like:

```python
def check_dir_exist(input_location=None, create=True):
    ...
    if not p.exists():
        if create:
            p.mkdir(parents=True)
    ...
```

now, we can do something like this

```python
@classmethod
def path(cls):
    conf_dir = check_dir_exist(CONF_DIR, create=cls.writable)
    return Path(conf_dir, DEFAULT_USER_CONF)
```